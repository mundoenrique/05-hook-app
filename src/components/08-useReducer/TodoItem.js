import React from 'react';

export default function TodoItem({ todo, handleToggle, index, handleDelete }) {
	return (
		<li className="list-group-item" key={todo.id}>
			<p
				className={`p-todo ${todo.done && 'complete'}`}
				onClick={() => handleToggle(todo.id)}
			>
				{index + 1} {todo.desc}
			</p>
			<button className="btn btn-danger" onClick={() => handleDelete(todo.id)}>
				Eliminar
			</button>
		</li>
	);
}
