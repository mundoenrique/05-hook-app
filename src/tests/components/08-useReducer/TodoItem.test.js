import { shallow } from 'enzyme';
import TodoItem from '../../../components/08-useReducer/TodoItem';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Pruebas sobre el componente <TodoItem />', () => {
	const handleToggle = jest.fn();
	const handleDelete = jest.fn();
	const indexArray = 0;

	let wrapper = shallow(
		<TodoItem
			todo={demoTodos[indexArray]}
			index={demoTodos.indexOf(demoTodos[indexArray])}
			handleDelete={handleDelete}
			handleToggle={handleToggle}
		/>
	);

	/* beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(
			<TodoItem
				todo={demoTodos[indexArray]}
				index={demoTodos.indexOf(demoTodos[indexArray])}
				handleDelete={handleDelete}
				handleToggle={handleToggle}
			/>
		);
	}); */

	test('Debe mostrarse correctamente <TodoItem />', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe llamar a la función handleDelete', () => {
		wrapper.find('.btn-danger').simulate('click');

		expect(handleDelete).toBeCalled();
		expect(handleDelete).toHaveBeenCalledTimes(1);
		expect(handleDelete).toHaveBeenCalledWith(demoTodos[indexArray].id);
	});

	test('Debe llamar a la función handleToggle', () => {
		wrapper.find('.p-todo').simulate('click');

		expect(handleToggle).toBeCalled();
		expect(handleToggle).toHaveBeenCalledTimes(1);
		expect(handleToggle).toHaveBeenCalledWith(demoTodos[indexArray].id);
	});

	test('Debe mostrar el texto correspondiente', () => {
		const desc = wrapper.find('.p-todo').text().trim();

		expect(desc).toBe(`${indexArray + 1} ${demoTodos[indexArray].desc}`);
	});

	test('Debe mostrar el todo completado (Class="complete")', () => {
		demoTodos[indexArray].done = true;

		wrapper = shallow(
			<TodoItem
				todo={demoTodos[indexArray]}
				index={demoTodos.indexOf(demoTodos[indexArray])}
				handleDelete={handleDelete}
				handleToggle={handleToggle}
			/>
		);

		const desc = wrapper.find('.p-todo');

		expect(desc.hasClass('complete')).toBe(true);
	});
});
