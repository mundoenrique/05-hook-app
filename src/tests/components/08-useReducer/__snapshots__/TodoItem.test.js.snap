// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <TodoItem /> Debe mostrarse correctamente <TodoItem /> 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="p-todo false"
    onClick={[Function]}
  >
    1
     
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Eliminar
  </button>
</li>
`;
